src/librustc_trans/mir/statement.rs

```diff
@@ -57,6 +58,22 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                     self.trans_rvalue(bcx, tr_dest, rvalue, debug_loc)
                 }
             }
+            mir::StatementKind::StorageLive(ref lvalue) => {
+                if let Some(index) = self.mir.local_index(lvalue) {
+                    if let LocalRef::Lvalue(tr_lval) = self.locals[index] {
+                        base::Lifetime::Start.call(&bcx, tr_lval.llval);
+                    }
+                }
+                bcx
+            }
+            mir::StatementKind::StorageDead(ref lvalue) => {
+                if let Some(index) = self.mir.local_index(lvalue) {
+                    if let LocalRef::Lvalue(tr_lval) = self.locals[index] {
+                        base::Lifetime::End.call(&bcx, tr_lval.llval);
+                    }
+                }
+                bcx
+            }
```

Comment:
These look like good candidates for helper methods.

I also don't fully understand why we cannot just called trans_lvalue?  I think it has a no-op case for locals?